### PR TITLE
fix label selector in jmp create lease

### PR DESCRIPTION
**Pull request by mangelajo**
our API receives selectors through a labels.Selector, which can parse things like "a=b, x!=3", but when storing to a lease label selector we use metav1.LabelSelector, that has its own parser, and does not support parsing "x!=3".

k8s does not seem to provide a label.Selector to -> metav1.LabelSelector, so we use the labels.Selector string parser, and convert the result into a metav1.LabelSelector by our own means.

Fixes-Issue: jumpstarter-dev/jumpstarter#135

## Summary by CodeRabbit

* **New Features**
  * More robust label selector parsing with support for != and broader operator handling for expressive selection.

* **Bug Fixes**
  * Improved validation and clearer error reporting for invalid or unsupported selector patterns to prevent misinterpretation.

* **Tests**
  * Added comprehensive tests covering simple, complex, negation, inclusion, existence, edge cases, and round‑trip selector behavior.

### CODE REVIEW

File: api/v1alpha1/lease_helpers.go [hunk 1]
```diff
@@ -73,12 +74,107 @@ func ReconcileLeaseTimeFields(beginTime, endTime **metav1.Time, duration **metav
...
+		case selection.Equals:
+			// For exact match with single value, use matchLabels
+			if len(values) == 1 {
+				// Check if we already have an equality requirement for this key with a different value
+				if existingValue, exists := matchLabels[key]; exists && existingValue != values[0] {
+					return nil, fmt.Errorf("invalid selector: label %s cannot have multiple equality requirements with different values (%s and %s)", key, existingValue, values[0])
+				}
+				matchLabels[key] = values[0]
+			} else {
+				// Multiple values should use In operator
```
Comment (bennyz) [inline api/v1alpha1/lease_helpers.go]: possibly redundant
https://pkg.go.dev/k8s.io/apimachinery/pkg/labels#NewRequirement
> If the operator is Equals, DoubleEquals, or NotEquals, the values set must contain one value.

Comment (mangelajo) [reply api/v1alpha1/lease_helpers.go]: Ahh no, while looking at this code it seems correct. the branch in L108 handles repeated keys, it will fail with error: invalid selector: label %s cannot have multiple equality requirements with different values (%s and %s)"

The 2nd branch handles the case where

a=[X,Y,Z],

not sure how you write it in their parseable format, but it means that a can be X, Y or Z, so they use an "a in (X, Y, Z)" selector.

I think we are fine ! :) please have a look.

**Review by bkhizgiy — APPROVED** [review]
Reviewed the request. It’s needed right now to fix a bug that’s affecting the lab. From what I see it looks good to me, hopefully I didn’t miss anything :)
